src/com/reco1l/osu/hud/elements/HUDNotesPerSecondCounter.kt

```diff
@@ -0,0 +1,48 @@
+package com.reco1l.osu.hud.elements
+
+import com.reco1l.andengine.text.ExtendedText
+import com.reco1l.osu.hud.HUDElement
+import com.reco1l.osu.playfield.SpriteFont
+import com.rian.osu.beatmap.hitobject.HitObject
+import kotlin.collections.ArrayDeque
+import ru.nsu.ccfit.zuev.osu.GlobalManager.getInstance as getGlobal
+import ru.nsu.ccfit.zuev.osu.ResourceManager
+import ru.nsu.ccfit.zuev.skins.OsuSkin
+
+class HUDNotesPerSecondCounter : HUDElement() {
+
+    override val name = "Notes per second counter"
+
+    private val label = ExtendedText().apply {
+        font = ResourceManager.getInstance().getFont("smallFont")
+        text = "Notes/sec"
+    }
+
+    private val value = SpriteFont(OsuSkin.get().scorePrefix).apply {
+        text = "0"
+    }
+
+    private val startTimes = ArrayDeque<Double>()
```

Comment:
`ArrayDeque<Double>` uses boxed `double` which it's kinda a concern when it comes to memory usage.
Can we use another thing such as a primitive double array?

Reply:
Changed to a list of objects so that it will use pointers rather than a boxed primitive.